tester/runner.go

```diff
@@ -274,9 +285,18 @@ func (r *Runner) runTests(ctx context.Context, txn storage.Transaction, enablePr
 		}
 	}
 
+	// Add custom builtins declarations to compiler
+	capabilities := ast.CapabilitiesForThisVersion()
+	for _, builtin := range r.customBuiltins {
+		capabilities.Builtins = append(capabilities.Builtins, builtin.Decl)
+	}
+
 	if r.compiler == nil {
 		r.compiler = ast.NewCompiler().
+			WithCapabilities(capabilities).
 			WithEnablePrintStatements(enablePrintStatements)
+	} else {
+		r.compiler.WithCapabilities(capabilities)
```

Comment:
Adding the capabilities to the compiler if none is provided seems reasonable.. not as sure when about the `else` clause here though.. if one provides a compiler, I would assume that they expect that compiler to be used exactly as it was provided... this essentially means that there is no way of running the tests _without_ extra capbabilities if those are found, no? I'm not sure if one would ever want that though :)

@srenatus WDYT?

Reply:
you're right. I must correct this

Reply:
Thanks! I take it your use case involves running the tests programmatically via the Go API and not just with `opa test`, is that correct?

Reply:
yes

Reply:
Indeed, I was thinking about how to do this with `opa test`... but since using custom builtins requires a custom build of OPA anyway, it would be trivial to add this to the test runner for that case.. so it's fine leaving this for now.

Reply:
i know that two similar if on compiler are not so elegant, but i consider also the case that the user sets the own compiler and decides to add custom builtins to it. Maybe an alternative can be this:
`	if r.compiler == nil {
		capabilities := ast.CapabilitiesForThisVersion()

		for _, builtin := range r.customBuiltins {
			capabilities.Builtins = append(capabilities.Builtins, builtin.Decl)
		}

		r.compiler = ast.NewCompiler().
			WithCapabilities(capabilities).
			WithEnablePrintStatements(enablePrintStatements)
	} else {
		capabilities := r.compiler.Capabilities()

		for _, builtin := range r.customBuiltins {
			capabilities.Builtins = append(capabilities.Builtins, builtin.Decl)
		}

		r.compiler.WithCapabilities(capabilities)
	}`

Reply:
If the user sets their own compiler, they can do what's in the else clause themselves, can't they?

```go
compiler := ast.NewCompiler()
capabilities := compiler.Capabilities()

for _, builtin := range r.customBuiltins {
	capabilities.Builtins = append(capabilities.Builtins, builtin.Decl)
}

compiler := compiler.WithCapabilities(capabilities)

runner.SetCompiler(compiler)
```

Reply:
okok